### DISCUSSION TITLE
Assign values directly to the store

### DISCUSSION THREAD

**Discussion by eliganim** [pmndrs/zustand#1120] [Q&A] [answered]
What are the downsides to assigning values directly to properties in the store, without using a setter?
For example:

`const useMyStore = create((set) => ({ value: 0 }))`

and then

```
const myStore = useMyStore()
myStore.value = 1
```

It appears to be working, but I assume it's wrong to do it this way?

**Comment by dai-shi** [comment, marked as answer]
It's not reactive, and may not work as expected in some cases.

It's mostly like you are doing this:
```js
const Component = () => {
  const [state, setState] = useState({ value: 0 })
  state.value = 1
  return <>{state.value}</>
}
```
